sdk/core/core-http/samples/gotHttpClient.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @summary This sample demonstrates how to implement a custom HttpClient.
+ */
+
+import { IncomingHttpHeaders } from "http";
+import * as coreHttp from "@azure/core-http";
+import got from "got";
+import { Headers as GotHeaders } from "got";
+
+/**
+ * Translate from core-http headers to got headers
+ * @param headers
+ * @returns
+ */
+function toOutgoingHeaders(headers: coreHttp.HttpHeadersLike): GotHeaders {
+  let result: Record<string, string> = {};
```

Comment:
You can replace this entire function with calling either `rawHeaders()` or `toJson()`: https://github.com/Azure/azure-sdk-for-js/blob/cf85b3e7846376ff41fedb0435546a13d376d71e/sdk/core/core-http/src/httpHeaders.ts#L61